client/src/Movies/Movie.js

```diff
@@ -27,17 +57,19 @@ export default class Movie extends Component {
         console.error(error);
       });
   };
+  
   // Uncomment this code when you're ready for the stretch problems
-  // componentWillReceiveProps(newProps){
-  //   if(this.props.match.params.id !== newProps.match.params.id){
-  //     this.fetchMovie(newProps.match.params.id);
-  //   }
-  // }
 
-  // saveMovie = () => {
-  //   const addToSavedList = this.props.addToSavedList;
-  //   addToSavedList(this.state.movie)
-  // }
+  componentWillReceiveProps(newProps){
+    if(this.props.match.params.id !== newProps.match.params.id){
+      this.fetchMovie(newProps.match.params.id);
+    }
+  }
+
+  saveMovie = () => {
+    const addToSavedList = this.props.addToSavedList;
+    addToSavedList(this.state.movie)
+  }
```

Comment:
Challenge: Right now the `addToSavedList` adds a movie to the savedList even though the movie is already on the list. Can you think of a way to refactor `addToSavedList` so that it will only add the movie if it has been added before?